benchmarks/benchmarks/stats.py

```diff
@@ -56,7 +56,99 @@ def time_ttest_ind_diff_var(self):
         stats.ttest_ind(self.a, self.c, equal_var=False)
 
 
+class DistributionsAll(Benchmark):
+    # all distributions are in this list. A conversion to a set is used to
+    # remove duplicates that are files under `distcont` and `distdiscrete`.
+    dists = sorted(list(set([d[0] for d in distcont + distdiscrete])))
+
+    param_names = ['distribution', 'properties']
+    params = [
+        dists, ['pdf/pmf', 'logpdf/logpmf', 'cdf', 'logcdf', 'rvs', 'fit',
+                'sf', 'logsf', 'ppf', 'isf', 'moment', 'stats_s', 'stats_v',
+                'stats_m', 'stats_k', 'stats_mvsk', 'entropy']
+    ]
+    dist_data = dict(distcont + distdiscrete)
+    # custom shape values can be provided for any distribution in the format
+    # `dist_name`: [shape1, shape2, ...]
+    custom_input = {}
+
+    # these are the distributions that are the slowest
+    slow_dists = ['nct', 'ncx2', 'argus', 'cosine', 'foldnorm', 'gausshyper',
+                  'kappa4', 'invgauss', 'wald', 'vonmises_line', 'ksone',
+                  'genexpon', 'exponnorm', 'recipinvgauss', 'vonmises',
+                  'foldcauchy', 'kstwo', 'levy_stable', 'skewnorm']
+    slow_methods = ['moment']
+
+    def setup(self, distribution, properties):
+        if not is_xslow() and (distribution in self.slow_dists or
+                               properties in self.slow_methods):
+            raise NotImplementedError("Skipped")
+
+        self.dist = getattr(stats, distribution)
+
+        shapes = self.dist_data[distribution]
+
+        if isinstance(self.dist, stats.rv_discrete):
+            # discrete distributions only use location
+            self.isCont = False
+            kwds = {'loc': 4}
+        else:
+            # continuous distributions use location and scale
+            self.isCont = True
+            kwds = {'loc': 4, 'scale': 10}
+
+        bounds = self.dist.interval(.99, *shapes, **kwds)
+        x = np.linspace(*bounds, 100)
+        args = [x, *self.custom_input.get(distribution, shapes)]
+
+        if properties == 'fit':
+            # the only positional argument is the data to be fitted
+            self.args = self.dist.rvs(size=100, **kwds)
```

Comment:
Come to think of it, you created a separate benchmark set for fitting, right? So we can get rid of this, too.

Reply:
There is a separate distribution that extensively tests fitting, but not for all distributions. Only the ones we overrode the fit methods for. 

I left this in since it tests all of them (though less extensively). We could have the fit benchmark test them all instead of here?

Reply:
Yeah I suppose they have different purposes. I envision these benchmarks here as helping us identify the methods that would benefit most from being overridden. The `fit` benchmarks are for tracking `fit` method performance over time for those that have been overridden. OK to keep this, but you should include the shape parameters in the call to `rvs`.

Reply:
I just noticed that as well. It's cumbersome to run them all but I am now running a only a few distributions to make sure it works.

Reply:
So they need to be included, or there would be an error.